### DISCUSSION TITLE
Error importing pref file on other device

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#311] [Q&A] [answered]
Last version .32 exported pref file leads to an error trying to import on the other device Samsung A52s.
When I only try to import all favorites, I get another error.

<img src="https://github.com/devemux86/cruiser/assets/121371629/7927b0ff-e932-4c73-b68b-5369fc7bcbb0" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/121371629/7fcc9cce-2c4b-46e5-af5a-9ca7346052a9" width="50%"/>

**Comment by devemux86** [comment]
@DolomitesTom last version does not export the old pref format. It exports the new json format.
Also the new profile manager works only with the new json format.

There will be for some versions the "Settings | Application | Advanced | Preferences | Import"
to import old pref files, then we must use the new profile manager for everything.

**Reply by DolomitesTom** [reply]
I do not understand. When I export all the profile manager, then a zip file is created. Do I need to unzip this for importing?

**Reply by devemux86** [reply, marked as answer]
> Do I need to unzip this for importing?

When Android file browser opens it shows a message with the expected file format: gpx, json, ...

So we must unzip the archives and then import them.
(we can select multiple files)

**Comment by devemux86** [comment]
@DolomitesTom

Please attach the files you are trying to import so we can check them.
Also mention the steps you are doing to import the files.

(the old pref files can only be imported in Settings)

**Reply by DolomitesTom** [reply]
Sorry but this is too personal. I don't want to put all settings including favorites here
 Hope you understand this.

**Reply by devemux86** [reply]
> this is too personal. I don't want to put all settings including favorites here

For the favorites it is understandable. :+1: 

However, the profile files do not contain any personal information, only the app settings.

**Reply by DolomitesTom** [reply]
Profiles is working now thanks. How do I import all favorites together? Or do I also need to import each single category? What about the symbols and colors of the categories?

**Reply by devemux86** [reply]
@DolomitesTom 

It is a similar process:
- Use "Export all" in one device (it exports a zip with all favorites with groups / icons / colors)
- Unzip the file in the other device
- Use "Import" and multiple select all files (it will restore all groups / icons / colors)

**Comment by devemux86** [comment]
I might introduce import from zip, it's more complicated, I'll put it on my list.

**Reply by bios4** [reply]
Importing a zip which was created using "export all" from device A to device B without the need of unzipping the contents first would be highly appreciated!

**Comment by DolomitesTom** [comment]
Favorites is working too, also symbols and colors are kept, perfect, thanks a lot

**Comment by devemux86** [comment]
By the way, @DolomitesTom nice "App color"! :+1:

**Comment by Ciutnik** [comment]
@DolomitesTom 
Regarding the .pref file import error: I experienced the same thing when the .pref didn't export properly (it was suspiciously small in size). Try importing your file on another device, or even on the same one you exported from, to verify that the .pref file is correct.
By the way, I have a very similar App color 🙂

<img src="https://github.com/devemux86/cruiser/assets/122647782/0063ed34-153c-4721-8884-d9f0c1426fad" width="75%"/>

**Reply by DolomitesTom** [reply]
It works ... and the good thing is, the color can be changed to fit the bike color 😁, when using the BMW to blue, gray or red, KTM to orange, Honda to red (AT tricolor) or olive green (military look of the Crossi) 😉 ... or just another daily mood color, long live the individualisation.

**Reply by devemux86** [reply]
> or just another daily mood color, long live the individualisation.

:+1: 

These days I also use dark app theme with bright green color (195, 255, 0), like the image [here](https://github.com/devemux86/cruiser/discussions/310#discussion-5615642).
But I like to change the app color often, depending on the mood! :slightly_smiling_face:
